[Kindergarten Garden] Fix a mention of dynamic programming. (#1557)
exercism/ruby: exercises/practice/kindergarten-garden/.docs/instructions.append.md

# Instructions append

> The exercise requires you to write a class that provides a single method for each child in the kindergarten, respectively.

Reviewer: ```suggestion
The exercise requires you to write a class that provides a method for each child in the kindergarten.
```

> You can make your work easier and less repetitive if, instead of manually defining 12 different methods, you use a bit of metaprogramming.

Reviewer: ```suggestion
  You can make your work less repetitive (and more fun!) if, instead of manually defining 12 different methods, you use a bit of metaprogramming.
```

Author: I have no problems with this and the above suggestions. Are you, by the way, OK with amending the commit even for such more substantial changes, or do you prefer a dedicated new commit?

Reviewer: It's probably better to have a different PR for bigger changes. These appends should be small hints really, not explanations about how to do things though, so I wouldn't expect the text to be too long here :)

> Your solution could use the method `#define_method` (or `#define_singleton_method`), or alternatively a combination of `#method_missing` and `#respond_to_missing?`.

Reviewer: ```suggestion
If you're not sure where to start, explore the [`method_missing`](https://www.leighhalliday.com/ruby-metaprogramming-method-missing) or [`define_method`](http://rubymonk.com/learning/books/2-metaprogramming-ruby/chapters/25-dynamic-methods/lessons/72-define-method) methods.
```

Author: I have no problem with the rephrasing, but I kinda dislike the fact that the `method_missing` article overrides `respond_to?` (and only relatively late in the text) instead of `respond_to_missing?`, which would be more safe and idiomatic, as I am lead to believe (although I am by no means an expert on Ruby).

Problem is, I cannot find an article that would explain it more (only those that already presume the reader knows what `method_missing` is, e.g. https://thoughtbot.com/blog/always-define-respond-to-missing-when-overriding).

Reviewer: I still consider `respond_to_missing?` to be the new kid on the block, although you're very right :) 

OK, Let's stick with these links for now and as soon as we have better Concept pages on Exercism for these in Ruby, we'll switch to them.

@meatball133 I don't find the Ruby docs to be clear enough for Ruby newbies really. I'd rather like to an explanation tutorial/blog (ideally our concept pages!)

# Instructions append

The exercise requires you to write a class that provides a single method for
each child in the kindergarten, respectively. You can make your work easier and
less repetitive if, instead of manually defining 12 different methods, you use
a bit of metaprogramming. Your solution could use the method `#define_method`
(or `#define_singleton_method`), or alternatively a combination of

> `#method_missing` and `#respond_to_missing?`.

Author: Done!
